Check/Codebase/GitContributions.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * @file
+ * Contains \SiteAudit\Check\Codebase\GitContributions.
+ */
+
+/**
+ * Class SiteAuditCheckCodebaseGitContributions.
+ */
+class SiteAuditCheckCodebaseGitContributions extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('Git Contributions');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Determine lines of code added and removed by each contributor.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {
+    if (isset($this->registry['not_git'])) {
+      return dt('Site is not under git source control');
+    }
+    if (isset($this->registry['git_remote'])) {
+      return dt("The repository has whole of drupal's git history. Reporting contrinutions of users will not be helpful in this case.");
```

Comment:
`return dt('The repository contains the entire Drupal contribution history and will be ignored.');`